File: db4on/core/src/core/com/db4o/QEGreater.cs
namespace com.db4o
{
	/// <exclude></exclude>
	public class QEGreater : com.db4o.QEAbstract
	{
		internal override bool Evaluate(com.db4o.QConObject a_constraint, com.db4o.QCandidate
			 a_candidate, object a_value)
		{
			if (a_value == null)
			{
				return false;
			}
			return a_constraint.GetComparator(a_candidate).IsGreater(a_value);
		}

		public override void IndexBitMap(bool[] bits)
		{
			bits[com.db4o.inside.ix.IxTraverser.GREATER] = true;
		}
	}
}

File: db4oj/core/src/com/db4o/internal/query/processor/QEGreater.java
/* Copyright (C) 2004 - 2007  db4objects Inc.   http://www.db4o.com */

package com.db4o.internal.query.processor;

import com.db4o.internal.*;
import com.db4o.internal.handlers.*;

/**
 * @exclude
 */
public class QEGreater extends QEAbstract {

    boolean evaluate(QConObject constraint, QCandidate candidate, Object obj) {
        if (obj == null) {
            return false;
        }
        Comparable4 comparator = constraint.getComparator(candidate);
        if (comparator instanceof ArrayHandler) {
            return ((ArrayHandler) comparator).isGreater(obj);
        }
        return comparator.compareTo(obj) > 0;
    }

    public void indexBitMap(boolean[] bits) {
        bits[QE.GREATER] = true;
    }
}
